source/base/subscriptor.cc

```diff
@@ -151,14 +156,14 @@ Subscriptor::operator=(Subscriptor &&s) noexcept
 void
 Subscriptor::subscribe(const char *id) const
 {
-#ifdef DEBUG
   if (object_info == nullptr)
     object_info = &typeid(*this);
   ++counter;
 
-  // This feature is disabled when we compile with threads: see the
-  // documentation of this class.
-#  ifndef DEAL_II_WITH_THREADS
+#ifdef DEAL_II_WITH_THREADS
+  std::lock_guard<std::mutex> lock(mutex);
+#endif
```

Comment:
I think this should go above the assignment to `object_info` in case two threads try to write to that pointer at the same time. Is there a reason that it is below that I am missing?

Alternatively, we could make `object_info` atomic.

Reply:
I think that you might be right. I've fixed this now.